list_items.rb

```diff
@@ -0,0 +1,37 @@
+class List
+  def initialize(people, books, rentals)
+    @books = books
+    @people = people
+    @rentals = rentals
+  end
+
+  def list_of_people
+    puts 'No people found!' if @people.empty?
+
+    @people.each_with_index do |person, i|
+      puts "#{i})  Name: #{person.name}, Age: #{person.age}, ID: #{person.id}"
+    end
+  end
+
+  def list_of_books
+    puts 'No Book found!' if @books.empty?
+
+    @books.each_with_index do |book, i|
+      puts "#{i}) Title: #{book.title}, Author: #{book.author} "
+    end
+  end
+
+  def list_of_rentals
+    puts 'Select id of any person'
+    @people.each { |person| puts "id: #{person.id}, Person: #{person.name}" }
+
+    print 'Person id: '
+    person_id = gets.chomp.to_i
+
+    @rentals.each do |rental|
+      if rental.person.id.to_i == person_id
+        puts "Date: #{rental.date}, Book: '#{rental.book.title}' by #{rental.book.author}"
+      end
+    end
+  end
```

Comment:
Similarly, let's ensure that our methods follow the single responsibility principle by placing all requests for input and input validation in the `main.rb` file and/or classes/methods whose responsibilities are to request or validate input. Kindly read further on the same via [this link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid)

Reply:
@rloterh Thank you for the review. I've implemented the changes you requested for. All requests for inputs and input validation are now in a separate class called Reader. The create_item class and list_item class now have obey Single Responsibility principle